src/pkgcheck/checks/python.py

```diff
@@ -193,6 +193,47 @@ def check_depend(self, depend, flags, prefix):
                 return True
         return False
 
+    def check_pep517_depend(self, item):
+        """Check Python ebuilds for missing optional dependencies.
+
+        The problematic case for us is DISTUTILS_OPTIONAL,
+        DISTUTILS_USE_PEP517 != standalone, but no ${DISTUTILS_DEPS} anywhere.
+        """
+        has_distutils_optional = None
+        has_distutils_pep517_non_standalone = None
+
+        for var_node, _ in bash.var_assign_query.captures(item.tree.root_node):
+            var_name = item.node_str(var_node.child_by_field_name('name'))
+
+            if var_name == "DISTUTILS_OPTIONAL":
```

Comment:
If you're only covering pep517 case, then DISTUTILS_OPTIONAL doesn't really matter (unless you're aiming for optimization). I mean, in non-optional case they will obviously be present, so the check will succeed.

But then, if we're optimizing, then we should probably start by eliminating PYTHON_DEPS check for non-optional case since it's more expensive than this. But I don't think it's a real problem right now.

That said, I suppose the original code misses checking for PYTHON_DEPS in BDEPEND when using distutils-r1. I think the original intent was that for "pure" use of single-r1 RDEPEND may be sufficient (e.g. when you're just installing a Python script) — but for a real Python build system, you want PYTHON_DEPS as well.

Perhaps you could check for both during a single iteration of deps.

Reply:
You mean DISTUTILS_OPTIONAL but not PEP517? I'm inclined to just say we leave it given it's on the way out.